new/reclassify_amazon_linux_platform_family.md

```diff
@@ -0,0 +1,30 @@
+---
+RFC: unassigned
+Author: Sean OMeara <sean@chef.io>
+Status: Draft
+Type: Standards
+---
+
+# Title
+
+Reclassify Amazon Linux platform_family
+
+## Motivation
+
+   As time passes, Amazon Linux looks less and less like a RHEL clone.
+   Release version numbers do not match EL5, EL6, EL7.
+   Amazon is now a rolling release
+   Different package sets are available
+   Package names and contents are different
+   Service providers are different
+
+## Specification
+
+Change Amazon Linux platform_family from 'rhel' to 'amazon'
```

Comment:
If we make Amazon Linux 's family "fedora", we'll just have to start
putting tons of exception logic around platform_family 'fedora' instead of
"rhel".


On Sun, Jul 12, 2015 at 10:57 PM, Lamont Granquist <notifications@github.com
> wrote:

> In new/reclassify_amazon_linux_platform_family.md
> <https://github.com/chef/chef-rfc/pull/109#discussion_r34430724>:
>
> > +# Title
> > +
> > +Reclassify Amazon Linux platform_family
> > +
> > +## Motivation
> > +
> > +   As time passes, Amazon Linux looks less and less like a RHEL clone.
> > +   Release version numbers do not match EL5, EL6, EL7.
> > +   Amazon is now a rolling release
> > +   Different package sets are available
> > +   Package names and contents are different
> > +   Service providers are different
> > +
> > +## Specification
> > +
> > +Change Amazon Linux platform_family from 'rhel' to 'amazon'
>
> Amazon using sysvinit and Yum is just like old fedora, that isn't a
> significant difference. The Provider Resolver code can also just handle
> that and pickup the RedHat service provider or the systemd provider based
> on duck-typing the operating system. We should also duck-type Yum-vs-DNF as
> well to make those differences transparent and not require hand-wiring up
> lots of platform-version switching logic.
>
> And package repo compatibility isn't necessarily expected at the family
> level (e.g. "debian" vs. "ubuntu" both being in the same "debian" family).
> That /is/ what makes the platform_family of 'rhel' work, but we don't
> expect it anywhere else. And they're still both RPM based distrubutions
> with basically the same layout of config files as RHEL/Fedora, which makes
> it Fedora-derived.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/chef-rfc/pull/109/files#r34430724>.
>